app/controllers/choose/teams_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Choose::TeamsController < ApplicationController
+  before_action :authorization
+
+  def show
+    @teams = current_user.teams
+  end
+
+  def update
```

Comment:
What you think about reusing the current code to switch teams?

https://github.com/Codeminer42/cm42-central/blob/master/app/controllers/teams_controller.rb#L6-L15

And seems that with your changes you can remove the `if current_user` at line 7.